### ISSUE TITLE
New Rule: no-spread-args

### ISSUE THREAD

**Issue by simondean** [eslint/eslint#16407]
### Rule details

Prevent the array spread operator being used to pass method args with unspecified size

### Related ECMAScript feature

Array spread operator

### What type of rule is this?

Warns about a potential problem

### Example code

```js
const someArrayWithNonFixedSize = someFunction();
someArray.push(...someArrayWithNonFixedSize);
```

See https://github.com/nodejs/node/issues/16870 for more info
```


### Why should this rule be in the core instead of a plugin?

I'll take your guidance on this one.  The issue is common to node.js.  May or may not affect other JS engines.  

### Participation

- [ ] I am willing to submit a pull request to implement this rule.

### Additional comments

_No response_
